wasm/src/record/record_plaintext.rs

```diff
@@ -92,6 +116,49 @@ mod tests {
         assert_eq!(record.gates(), 99);
     }
 
+    #[wasm_bindgen_test]
+    fn test_serial_number() {
+        let pk = PrivateKey::from_string("APrivateKey1zkpDeRpuKmEtLNPdv57aFruPepeH1aGvTkEjBo8bqTzNUhE").unwrap();
+        let record = RecordPlaintext::from_string(RECORD).unwrap();
+        let program_id = "token.aleo";
+        let record_name = "token";
+        let expected_sn = "4564977995400415519058823909143155627601970323571971278914520967771079582104field";
+        let result = record.serial_number_string(&pk, program_id, record_name);
+        assert!(result.is_ok());
+        assert_eq!(expected_sn, result.unwrap());
+    }
+
+    #[wasm_bindgen_test]
+    fn test_serial_number_can_run_twice_with_same_private_key() {
+        let pk = PrivateKey::from_string("APrivateKey1zkpDeRpuKmEtLNPdv57aFruPepeH1aGvTkEjBo8bqTzNUhE").unwrap();
+        let record = RecordPlaintext::from_string(RECORD).unwrap();
+        let program_id = "token.aleo";
+        let record_name = "token";
+        let expected_sn = "4564977995400415519058823909143155627601970323571971278914520967771079582104field";
+        assert_eq!(expected_sn, record.serial_number_string(&pk, program_id, record_name).unwrap());
+        assert_eq!(expected_sn, record.serial_number_string(&pk, program_id, record_name).unwrap());
+    }
+
+    #[wasm_bindgen_test]
+    fn test_serial_number_invalid_program_id_returns_err_string() {
+        let pk = PrivateKey::from_string("APrivateKey1zkpDeRpuKmEtLNPdv57aFruPepeH1aGvTkEjBo8bqTzNUhE").unwrap();
+        let record = RecordPlaintext::from_string(RECORD).unwrap();
+        let program_id = "not a real program id";
+        let record_name = "token";
+        let expected_value = "Invalid ProgramID specified".to_string();
+        assert_eq!(record.serial_number_string(&pk, program_id, record_name).err(), Some(expected_value.into()));
+    }
+
+    #[wasm_bindgen_test]
+    fn test_serial_number_invalid_record_name_returns_err_string() {
+        let pk = PrivateKey::from_string("APrivateKey1zkpDeRpuKmEtLNPdv57aFruPepeH1aGvTkEjBo8bqTzNUhE").unwrap();
+        let record = RecordPlaintext::from_string(RECORD).unwrap();
+        let program_id = "token.aleo";
+        let record_name = "not a real record name";
+        let expected_value = "Invalid Identifier specified for record".to_string();
+        assert_eq!(record.serial_number_string(&pk, program_id, record_name).err(), Some(expected_value.into()));
+    }
```

Comment:
These tests might be overkill, but I added to them to demonstrate the behavior. Wasn't sure how to do an equivalent test for a commitment failure though, as that's not as trivial as passing in an incorrect string.

Reply:
More test coverage is always better - happy to have further tests.